sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -951,6 +951,14 @@ object SQLConf {
     .checkValue(_ > 0, "the value of spark.sql.sources.bucketing.maxBuckets must be greater than 0")
     .createWithDefault(100000)
 
+  val AUTO_BUCKETED_SCAN_ENABLED =
+    buildConf("spark.sql.sources.bucketing.autoBucketedScan.enabled")
+      .doc("When true, decide whether to do bucketed scan on input tables based on query plan " +
```

Comment:
Since our user documents are generated based on this statement, could you describe a bit more about how to decide whether to do bucketed scans or not?

Reply:
@maropu - sure, wondering what do you think of below?

```
When true, decide whether to do bucketed scan on input tables
based on query plan automatically. Do not use bucketed scan if
(1).query does not have operators to utilize bucketing (e.g. join, group-by, etc),
or (2).there's an exchange operator between these operators and table scan.

```

Reply:
It looks okay, but I have the same suggestion with https://github.com/apache/spark/pull/29804#discussion_r494852129

Reply:
@maropu - sure, updated.